### PR TITLE
Fix hung Google token refresh permanently freezing calendar sync

### PR DESCRIPTION
**Pull request by GeoffreyPlitt**
### Status
**READY**

### Description

Fixes a reliability bug in the Google Calendar provider where a single hung OAuth token refresh permanently freezes all calendar refreshes with no error, no retry, and no user-visible signal.

This caused a real user-facing failure (observed on v4.11.6; the same pattern exists on master): token refreshes stopped silently after a sleep/wake cycle, the menu kept showing the previous day's events all greyed out (past-event styling), the status bar lost its countdown, and no notifications fired until the app was manually restarted.

**Root cause** (`GCEventStore.validAccessToken`): token refreshes are serialized through a stored `refreshTask`, and the task body wraps AppAuth's `OIDAuthState.performAction` callback in a continuation. If `performAction` never invokes its callback (observed in the wild — e.g. its internal token-endpoint request gets parked across sleep/wake), the continuation never resumes, `defer { refreshTask = nil }` never runs, and every subsequent fetch awaits the dead task forever. There was no timeout anywhere in this path. Additionally, the `guard let self else { return }` inside the callback could leak the continuation outright, and the same serialize-through-a-stored-task pattern in `ensureSignedIn`/`signInTask` had the same wedge risk.

**Fix**:

* New `TimeoutGuardedCompletion` helper (in `GoogleCalendarPolicy.swift`, part of the `MeetingBarLogic` package) that races a completion-callback-based operation against a hard timeout and guarantees the continuation resumes exactly once, even when the timeout and a late callback race.
* `validAccessToken` wraps `performAction` with a 30s timeout; on timeout it throws `AuthError.refreshFailed`, `refreshTask` is always cleared, and the existing `ProviderHealth` machinery surfaces the stale-data warning as it already does for thrown errors.
* `ensureSignedIn` gets the same protection with a 5-minute timeout (interactive sign-in can legitimately take a while), so a never-completing sign-in can no longer block all future refreshes.
* The `guard let self` continuation leak is gone — the AppAuth callback no longer touches `self`.

No public API changes; EventKit provider untouched.

## Checklist
- [ ] Localized (no user-facing strings added)
- [ ] Added to changelog — no unreleased section exists yet; happy to add entries once the target version is known:
  - [ ] [Changelog View](https://github.com/leits/MeetingBar/blob/master/MeetingBar/UI/Views/Changelog/Changelog.swift)
  - [ ] [CHANGELOG.md](https://github.com/leits/MeetingBar/blob/master/CHANGELOG.md)

### Steps to Test or Reproduce

1. `make test` — new unit tests in `MeetingBarLogicTests/TimeoutGuardedCompletionTests.swift` cover: a callback that never fires throws within the timeout instead of hanging; a late callback after timeout is safely ignored (no double-resume crash); a subsequent call is not blocked by an earlier hung operation; fast success/failure paths are unaffected.
2. Full CI (SwiftPM logic tests + app-hosted tests + lint) is green on the fork: 213 + 429 tests, 0 failures.
3. The hung-callback condition itself only reproduces on rare network/sleep-wake edge cases, so end-to-end verification was regression-focused: sign-in and calendar refresh flow through the new timeout wrapper unchanged.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Google Calendar sign-in now completes with a clear cancellation or timeout result instead of remaining indefinitely.
  * Token refresh operations are time-limited and report a refresh failure when they exceed the allowed duration.
  * Authentication state continues to be cleared appropriately when authorization is no longer valid.
* **Tests**
  * Added coverage for sign-in and token-refresh timeout behavior, delayed callbacks, failures, and subsequent operations.

### CODE REVIEW

File: MeetingBar/Calendar/Providers/Google/GoogleCalendarEventStore.swift [hunk 1]
```diff
@@ -288,11 +295,24 @@ final class GCEventStore: NSObject,
...
+                }
+                Task {
+                    do {
+                        try await self.signIn(forcePrompt: forceConsent)
+                        completion(.success(()))
+                    } catch {
+                        completion(.failure(error))
+                    }
+                }
+            }
```
[the remark this answers is not in the document]
Comment (GeoffreyPlitt) [reply MeetingBar/Calendar/Providers/Google/GoogleCalendarEventStore.swift:313]: Addressed in 96f6fdd, but with a different fix than suggested. Mapping the sign-in timeout to `AuthError.refreshFailed` would surface "Google Calendar token refresh failed" for an interactive sign-in that timed out — a mislabeled diagnostic, since no token refresh is involved on that path.

The consistency gap is narrower than it looks: `ProviderHealth.isAuthRequired` classifies both `AuthError.refreshFailed` and an unrecognized error as `authRequired: false`, so the menu state machine behaves identically either way. The only real difference was the user-visible message — `OperationTimedOut` fell through to Foundation's default "The operation couldn't be completed. (MeetingBar.OperationTimedOut error 1.)".

Fixed by conforming `OperationTimedOut` to `LocalizedError` ("Google Calendar request timed out"), which gives both guarded paths an accurate, readable message while keeping the error types semantically distinct.
